src/hdf5_interface.F90

```diff
@@ -1836,40 +1838,41 @@ subroutine read_string(buffer, obj_id, name, indep)
       if (indep) data_xfer_mode = H5FD_MPIO_INDEPENDENT_F
     end if
 
-    ! Get dataspace
-    call h5dget_space_f(dset_id, space_id, hdf5_err)
-
     ! Make sure buffer is large enough
     call h5dget_type_f(dset_id, filetype, hdf5_err)
-    call h5tget_size_f(filetype, size, hdf5_err)
-    if (size > len(buffer) + 1) then
+    call h5tget_size_f(filetype, n, hdf5_err)
```

Comment:
It is still possible to have strings that are too large to read, even with deferred length strings?